src/main/java/gui/viewmodel/MonthlyCalendarViewModel.java

```diff
@@ -1,38 +1,48 @@
 package gui.viewmodel;
 
 import com.calendarfx.model.Entry;
+import datagateway.Observer;
 import datagateway.event.EventReader;
-import datagateway.event.ObservableEventRepository;
 import gui.utility.EventHelper;
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
+import services.eventcreation.CalendarEventCreationBoundary;
+import services.eventcreation.EventSaver;
+import services.eventpresentation.CalendarEventRequestBoundary;
+import services.eventpresentation.EventInfo;
+import services.updateentities.UpdateEventBoundary;
 
+import java.util.ArrayList;
 import java.util.List;
 
-public class MonthlyCalendarViewModel extends ViewModel {
+public class MonthlyCalendarViewModel extends ViewModel implements Observer<EventReader> {
 
-    private final ObservableEventRepository repository;
+    private final CalendarEventCreationBoundary eventAdder;
+    private final CalendarEventRequestBoundary eventGetter;
+    private final UpdateEventBoundary eventUpdater;
+    private final EventSaver eventSaver;
 
     private final ObservableList<Entry<String>> entryList;
 
-    public MonthlyCalendarViewModel(ObservableEventRepository repository) {
-        this.repository = repository;
-        repository.addCreationObserver(this::handleCreation);
-        repository.addUpdateObserver(this::handleUpdate);
-
-        List<EventReader> eventReaderList = this.repository.getAllEvents();
-        this.entryList = FXCollections.observableArrayList(EventHelper.eventReaderToEntry(eventReaderList));
+    public MonthlyCalendarViewModel(CalendarEventCreationBoundary eventAdder,
+                                    CalendarEventRequestBoundary eventGetter,
+                                    UpdateEventBoundary eventUpdater,
+                                    EventSaver eventSaver) {
+        this.eventAdder = eventAdder;
+        this.eventGetter = eventGetter;
+        this.eventUpdater = eventUpdater;
+        this.eventSaver = eventSaver;
+        List<EventInfo> eventInfoList = new ArrayList<>(); // TODO: change this to be getting events from EventGetter
```

Comment:
How come you can't do this todo already? Is there some change that we need to do first

Reply:
got it done now! was waiting for another pr to be merged